open-apparel-registry/open-apparel-registry
# Enable administrators to split facility matches into new facilities

## Overview

This PR enables administrators to split facility matches into new facilities via a new `/dashboard/splitfacilities` route. The new page has the same facility card as is displayed on the merge and delete UIs, but also displays a "Matches to Split" card on the righthand side which displays details about each match along with a "Split" button used to open a confirmation dialog before creating a new facility. On successful splits, we update the UI to display the new OAR ID and also to filter the match out of the source facility's "Contributors" list.

Connects #593 

## Demo

![Screen Shot 2019-06-25 at 7 33 40 PM](https://user-images.githubusercontent.com/4165523/60140530-28f0f980-9780-11e9-8273-d0480f9c03fc.png)

## Notes

~Although this might be an edge case I noticed a bug in what I think should be a valid operation: after splitting a match from Facility A into Facility B, I attempted to merge Facility A into Facility B and encountered this error:~

```
("Cannot delete some instances of model 'Facility' because they are referenced through a protected foreign key: 'FacilityListItem.facility'", <QuerySet [<FacilityListItem: FacilityListItem 594 - MATCHED>, <FacilityListItem: FacilityListItem 277 - MATCHED>]>)
```

~Merging B back into A worked as expected, so I'm not quite sure what causes the error above. Any ideas?~

I believe I fixed this and that the problem was that I had neglected to update the facility_list_item.facility to the new facility.
## Testing Instructions

- serve this branch then `resetdb` & `processfixtures`
- sign in as c1 then search for facility name "Regina" in country "China" and copy the OAR ID
- visit the dashboard page, then click on the "Split Facility Matches" link
- search for the OAR you copied
- split the matches and verify that they create a new facility
- you can also try out merging the new facilities back into the original facility, or merging facilities then splitting them then merging them etc

## Checklist
